tests/system_test.py

```diff
@@ -1013,8 +1018,16 @@ def on_link_closing(self, event):
             rcv = event.container.create_receiver(event.connection,
                                                   **kwargs)
 
+    def queue_put(self, msg):
```

Comment:
You can avoid changing the API (and all those other files) by creating an internal class to the AsyncTestReceiver that inherits from Queue and overrides get() and put() to do what you want.

Something like:

class _MyQueue(Queue):                                                                                                                            
   def __init__(self, receiver):                                                                                                                  
       self._async_receiver = receiver                                                                                                            
       super(_MyQueue, self).__init__()                                                                                                           
                                                                                                                                                  
   def get(self, timeout=TIMEOUT):                                                                                                                
       self._async_receiver.num_queue_gets +=1                                                                                                    
       super(_MyQueue, self).get(timeout)                                                                                                         
                                                                                                                                                  
   def put(self, msg):                                                                                                                            
       self._async_receiver.num_queue_puts +=1                                                                                                    
       super(_MyQueue, self).put(msg)                                                                                                             
                                                                                                                                                  
                                                                                                                                                  
then in the async receiver's __init__:                                                                                                
    self.queue = _MyQueue(self)